### DISCUSSION TITLE
Collapsable left navigation bar

### DISCUSSION THREAD

**Discussion by cjlotz** [dotnet/docfx#10169] [Q&A]
Hello

Is there a feature/templates whereby the user can collapse/expand the left navigation bar to make more of the content horizontally available or not?

**Comment by filzrev** [comment]
>  user can collapse/expand the left navigation bar to make more of the content horizontally available

How about switching the `data-layout` attribute by JavaScript?

By default docfx use `data-layout=""` (`data-layout="conceptual"`) layout.
And this layout show top/left navigation bars

https://github.com/dotnet/docfx/blob/main/templates/modern/src/layout.scss#L129-L130.

And it can change  `data-layout`  attribute with following JavaScript code.

## Switch to `landing` layout
```
document.getElementsByTagName('body')[0].setAttribute('data-layout', 'landing')
```

## Switch to `chromeless` layout
```
document.getElementsByTagName('body')[0].setAttribute('data-layout', 'chromeless')
```

## Revert to `default`(`conceptual`) layout
```
document.getElementsByTagName('body')[0].setAttribute('data-layout', '')
or
document.getElementsByTagName('body')[0].setAttribute('data-layout', 'conceptual')
```
